internal/locate/region_request.go

```diff
@@ -2302,3 +2306,34 @@ func (s *staleReadMetricsCollector) onResp(tp tikvrpc.CmdType, resp *tikvrpc.Res
 		metrics.StaleReadRemoteInBytes.Add(float64(size))
 	}
 }
+
+func GetReadType(req *tikvrpc.Request) string {
```

Comment:
I suppose staleness and replica are two independent dimensions? For example shall we distinguish stale-follower-read and stale-leader-read?

Reply:
Do you mean there is 3 dimenstions:

- is retry
- staleness
- the selected replica is follower or leader(it does not depend on the `ReplicaReadType`)

Reply:
Yes, this approach seems clearer to me. The only concern is its verbosity, especially when we need to record the type of the last attempt.

Reply:
So it's look like:

- `leader` means request sent to leader first time(it can be either leader read or replica read)
- `follower` means request sent to follower first time(it should be replica read)
- `stale_leader` means stale read on leader replica first time.
- `stale_follower` means stale read on follower replica first time.
- `retry_stale_leader_leader` means fallback to leader read on leader replica from stale read on leader replica.
- `retry_stale_follower_leader` means fallback to leader read on leader replica from stale read on follower replica.
- `retry_stale_leader_follower` means fallback to replica read on follower replica from stale read on leader replica(the leader is exhausted).
- `retry_stale_follower_follower` means fallback to replica read on follower replica from stale read on follower replica(the leader is exhausted).